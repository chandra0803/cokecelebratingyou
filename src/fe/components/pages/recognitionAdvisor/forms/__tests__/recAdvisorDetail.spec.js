/*global jest*/

jest.mock( 'react-redux' );
import React from 'react';
import RecAdvisorDetail from '../recAdvisorDetail.js';
import renderer from 'react-test-renderer';


window.recognitionAdvisor = {
  raUrl: 'ra/reminders.action',
  raTilePageDisplay: 'yes',
  raDetailPageDisplay: 'no',
  raEndModelPageDisplay: 'no',
  content: []
};

const cmStringsArrayToObject = ( propertiesArray ) => {
    const propertiesObject = {};
    if( propertiesArray ) {

        propertiesArray.forEach( property => {
            propertiesObject[ `${ property.code }.${ property.key.toLowerCase() }` ] = property.content;
        } );
    }
    return propertiesObject;
};

const defaultProps = {
    recognitionAdvisor: {
      raUrl: 'ra/reminders.action',
      raTilePageDisplay: 'yes',
      raDetailPageDisplay: 'no',
      raEndModelPageDisplay: 'no',
      content: []
    }
};

const raDetailModule = {
  'rowNumStart': 1,
  'rowNumEnd': 50,
  'activePage': 0,
  'sortColName': 'NO_VALUE',
  'sortedBy': '',
  'excludeUpcoming': 0,
  'filterValue': 0,
  'pendingStatus': 0
};

it( 'renders the dom like the snapshot when the props have default values.', () => {

  const props = {
      ...defaultProps,
      recognitionAdvisorFunc: jest.fn(),
      raEligibleProgramsFunc: jest.fn()
  };
  const state = {
    attributeData: raDetailModule,
    raTilePage: true,
    raDetailPage: false,
    raEndModelPage: false,
    content: cmStringsArrayToObject( window.recognitionAdvisor.content )
  };
  const component = renderer.create(
    <div>
      <RecAdvisorDetail { ...state } { ...props } />
    </div>
  );

  const tree = component.toJSON();
  expect( tree ).toMatchSnapshot();
} );
