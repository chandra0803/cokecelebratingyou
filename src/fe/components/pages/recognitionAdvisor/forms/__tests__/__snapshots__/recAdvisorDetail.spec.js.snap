// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the dom like the snapshot when the props have default values. 1`] = `
<div>
  <div
    className="raDetailPage"
  >
    <div>
      <h3
        className="module-title"
      />
      <div
        className="ra-wrapper"
      >
        <div
          className="clearfix list-container col-count-2"
        />
      </div>
      <div
        className="module-actions"
      >
        <a
          href="undefined/raDetails.do"
        />
      </div>
    </div>
    <div>
      <div
        className="raInsights"
      >
        <div>
          <div
            className="raInsightsLogoPosition"
          >
            <div
              className="raInsightsLogo"
            >
              <svg
                data-name="Layer 1"
                id="Layer_1"
                viewBox="0 0 191.97 189.22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <defs>
                  <clipPath
                    id="clip-path"
                    transform="translate(0 0)"
                  >
                    <rect
                      height="189.21"
                      width="191.97"
                    />
                  </clipPath>
                </defs>
                <title>
                  Vector Smart Object
                </title>
                <g>
                  <path
                    className="raBrandingOuter"
                    d="M183.85,33a95.66,95.66,0,0,0-22-31.16,6.69,6.69,0,0,0-9.19,0A82.26,82.26,0,0,1,96,24.31,82.26,82.26,0,0,1,39.35,1.82a6.7,6.7,0,0,0-9.19,0,96,96,0,0,0,7.38,146h0l-8.9,41.4,54.95-22.35h0a96.83,96.83,0,0,0,12.41.8,95.91,95.91,0,0,0,96-96A95.26,95.26,0,0,0,183.85,33m-29.46,97.09A82,82,0,0,1,96,154.27a83.63,83.63,0,0,1-13.79-1.14L46.92,166.91,52.7,142a83.23,83.23,0,0,1-15.12-12A82.51,82.51,0,0,1,35.08,15.9,95.56,95.56,0,0,0,96,37.7a95.56,95.56,0,0,0,60.91-21.8,82.59,82.59,0,0,1-2.51,114.18"
                    transform="translate(0 0)"
                  />
                  <path
                    className="raBrandingInner"
                    d="M140.63,59.42a24.39,24.39,0,0,0-17,6.88,10.21,10.21,0,0,0-1,.83L96,92.2,69.39,67.13a10.1,10.1,0,0,0-1-.83,24.55,24.55,0,1,0,.32,35A24.74,24.74,0,0,0,72,97.22l17.07,16.09a10,10,0,0,0,13.78,0l17.07-16.09a24.74,24.74,0,0,0,3.32,4.11,24.55,24.55,0,1,0,17.36-41.92M51.34,95.14A11.16,11.16,0,1,1,62.5,84,11.16,11.16,0,0,1,51.34,95.14m89.29,0A11.16,11.16,0,1,1,151.79,84a11.16,11.16,0,0,1-11.16,11.16"
                    transform="translate(0 0)"
                  />
                </g>
              </svg>
            </div>
          </div>
          <div
            className="raInsightsBackgrondImg"
          />
          <div
            className="raInsightsGradgient"
          />
        </div>
        <div
          className="raInsightsBodyContent"
        >
          <h4 />
          <p />
          <div
            className="clear"
          />
        </div>
      </div>
    </div>
    <div>
      <h3 />
      <div
        className="raFilterWrap"
      >
        <i
          className="icon-arrow-1-down detailArrowToken"
        />
        <span
          className="filterLabel"
        />
        <select
          className="filterTokens"
          onChange={[Function]}
          value={0}
        >
          <option
            value="0"
          />
          <option
            disabled={false}
            value="1"
          />
          <option
            disabled={false}
            value="2"
          />
          <option
            disabled={false}
            value="3"
          />
        </select>
        <a
          className="raPromo fr"
          data-tip="React-tooltip"
        />
      </div>
      <div
        className="pagination"
      >
        <div
          className="counts"
        >
          <span
            className="start"
          />
          –
          <span
            className="end"
          />
           
          <em
            className="of"
          >
            of
          </em>
          <span
            className="total"
          >
             
             
          </span>
        </div>
        <ul
          className="pagination"
        >
          <li
            className="disabled"
            onClick={[Function]}
          >
            <a
              className=""
              href="#"
            >
              <i
                className="icon-double-arrows-1-left"
              />
            </a>
          </li>
          <li
            className="disabled"
            onClick={[Function]}
          >
            <a
              className=""
              href="#"
            >
              <i
                className="icon-arrow-1-left"
              />
            </a>
          </li>
          <li
            className="disabled"
            onClick={[Function]}
          >
            <a
              className=""
              href="#"
            >
              <i
                className="icon-arrow-1-right"
              />
            </a>
          </li>
          <li
            className=""
            onClick={[Function]}
          >
            <a
              className=""
              href="#"
            >
              <i
                className="icon-double-arrows-1-right"
              />
            </a>
          </li>
        </ul>
      </div>
      <div
        className="__react_component_tooltip place-top type-dark "
        data-id="tooltip"
      >
        
      </div>
      <div
        className="myteamTable"
      >
        <table
          className="table table-striped"
        >
          <thead>
            <tr>
              <th
                className="active"
                onClick={[Function]}
              >
                <a
                  data-sort="is_new_hire"
                  data-sortBy="desc"
                  href="#"
                  onClick={[Function]}
                >
                   
                </a>
                 
                <i
                  className="icon-arrow-1-down"
                />
              </th>
              <th
                className="null"
                onClick={[Function]}
              >
                <a
                  data-sort="first_name"
                  data-sortBy="asc"
                  href="#"
                  onClick={[Function]}
                />
                 
                <i
                  className="icon-arrow-1-up"
                />
              </th>
              <th
                className="null"
                onClick={[Function]}
              >
                <a
                  data-sort="by_me_date_sent"
                  data-sortBy="asc"
                  href="#"
                  onClick={[Function]}
                />
                 
                <i
                  className="icon-arrow-1-up"
                />
              </th>
              <th
                className="null"
                onClick={[Function]}
              >
                <a
                  data-sort="by_oth_date_sent"
                  data-sortBy="asc"
                  href="#"
                  onClick={[Function]}
                />
                 
                <i
                  className="icon-arrow-1-up"
                />
              </th>
              <th />
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
      <div
        className="pagination"
      >
        <div
          className="counts"
        >
          <span
            className="start"
          />
          –
          <span
            className="end"
          />
           
          <em
            className="of"
          >
            of
          </em>
          <span
            className="total"
          >
             
             
          </span>
        </div>
        <ul
          className="pagination"
        >
          <li
            className="disabled"
            onClick={[Function]}
          >
            <a
              className=""
              href="#"
            >
              <i
                className="icon-double-arrows-1-left"
              />
            </a>
          </li>
          <li
            className="disabled"
            onClick={[Function]}
          >
            <a
              className=""
              href="#"
            >
              <i
                className="icon-arrow-1-left"
              />
            </a>
          </li>
          <li
            className="disabled"
            onClick={[Function]}
          >
            <a
              className=""
              href="#"
            >
              <i
                className="icon-arrow-1-right"
              />
            </a>
          </li>
          <li
            className=""
            onClick={[Function]}
          >
            <a
              className=""
              href="#"
            >
              <i
                className="icon-double-arrows-1-right"
              />
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
